contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -252,30 +267,29 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
     ) public onlyWhenUnlocked {
         require(currentValidatorSetSize() < validatorThreshold, "Validator set Threshold exceeded!");
         require(amount > minDeposit, "min deposit limit failed!");
-        require(heimdallFee >= minHeimdallFee, "Minimum amount is 1 Matic");
-
-        require(token.transferFrom(msg.sender, address(this), amount.add(heimdallFee)), "Transfer stake failed");
-        _stakeFor(user, amount, acceptDelegation, signerPubkey);
-        // _topup
         _topUpForFee(user, heimdallFee);
+        require(token.transferFrom(msg.sender, address(this), amount), "Transfer stake failed");
+        _stakeFor(user, amount, acceptDelegation, signerPubkey);
     }
 
     function unstakeClaim(uint256 validatorId) public onlyStaker(validatorId) {
+        uint256 deactivationEpoch = validators[validatorId].deactivationEpoch;
         // can only claim stake back after WITHDRAWAL_DELAY
         require(
-            validators[validatorId].deactivationEpoch > 0 &&
-                validators[validatorId].deactivationEpoch.add(WITHDRAWAL_DELAY) <= currentEpoch &&
-                validators[validatorId].status != Status.Unstaked
+            deactivationEpoch > 0 && deactivationEpoch.add(WITHDRAWAL_DELAY) <= currentEpoch &&
+            validators[validatorId].status != Status.Unstaked
         );
+
         uint256 amount = validators[validatorId].amount;
-        totalStaked = totalStaked.sub(amount);
+        uint256 newTotalStaked = totalStaked.sub(amount);
+        totalStaked = newTotalStaked;
```

Comment:
how gas we've saved by one less `SLOAD`.

Reply:
Yes, worth it.